### DISCUSSION TITLE
Expose lighting information to the fragment stage

### DISCUSSION THREAD

**Discussion by PhantomFighter** [godotengine/godot-proposals#6215] [3D]
Apologies in advance if this is the wrong place for it.
It is fairly self explanatory. Unity support this out of the box. You have access to color, light vector, attenuation, etc in the fragment stage to do custom lighting/shading. We can't port these existing shaders over to Godot as it stands. Examples of some interesting projects include methods of Toon shaders, Eye shaders (of which I was working on one before I realized we can't access lighting in fragment).

Is this feasible to facilitate, or on the roadmap?

**Comment by Calinou** [comment]
This is already being tracked in https://github.com/godotengine/godot-proposals/issues/1620.
